service-userlayer/src/main/java/org/oskari/map/userlayer/service/UserLayerDataService.java

```diff
@@ -44,19 +50,29 @@
     private static final int USERLAYER_BASE_LAYER_ID = PropertyUtil.getOptional(USERLAYER_BASELAYER_ID, -1);
 
     public static UserLayer createUserLayer(SimpleFeatureCollection fc,
-            String uuid, String name, String desc, String source) {
+            String uuid, String name, String desc, String source)
+                    throws FactoryException, TransformException {
```

Comment:
I would wrap the exceptions to a ServiceException since (in my opinion) the details are not that interesting that a list of exceptions is required. A bunch of possible exceptions a method can throw tends to lead to a catch(Exception e) error handling.

I'm also not sure if problems when calculating the coverage/bounds for the data should be treated as a failure or should we just leave the WKT empty. Then again if the transform fails it could indicate a problem with the data and as such should be treated as a failure so we don't store invalid data.

Any thoughts on these?

Reply:
Yea this isn't nice. Previously this method didn't even throw any exceptions, so even an ServiceException seems like a downgrade. I'm not even sure if the FactoryException _can_ occur, since the throwing call is fixed CRS.decode("EPSG:4326", true) which shouldn't fail. For the TransformException I agree that _if_ for some reason it does fail there's probably something wrong with the data (don't know when this could even happen, maybe when coordinates are outside of the extent of the source projection? They won't be outside of WGS84 anyway, so that one probably exists mainly on the source projection side. 3D -> 2D maybe?)

Reply:
I'll make the changes